src/metabase/api/card.clj

```diff
@@ -129,6 +130,24 @@
   (-> (apply cards-for-filter-option* filter-option (when model-id-or-nil [model-id-or-nil]))
       (t2/hydrate :creator :collection)))
 
+(defn- hydrate-card-details
+  "Get card details to produce a complete view of the card"
+  [{card-id :id :keys [dataset] :as card}]
+  (span/with-span!
+    {:name       "hydrate-card-details"
+     :attributes {:card/id card-id}}
+    (-> card
+        (t2/hydrate :creator
+                    :dashboard_count
+                    :parameter_usage_count
+                    :can_write
+                    :average_query_time
+                    :last_query_start
+                    [:collection :is_personal]
+                    [:moderation_reviews :moderator_details])
+        (cond-> dataset (t2/hydrate :persisted))
+        collection.root/hydrate-root-collection)))
```

Comment:
Are you sure the root collection needs to be hydrated on all endpoints? I would expect that all collections are normalized on the frontend and cached separately from the card stuff, including the root collection.

Reply:
Agreed. Done.

Reply:
Per our discussion, let's go with removing. This means we'll not have _identical_ results as POST|PUT + GET, but the assumption is that the `:collection` will be present already on the FE in those cases.